In [1]:
import pandas as pd
import numpy as np
import statsmodels.api as sm

In [2]:
d1=pd.read_stata('oregonhie_descriptive_vars.dta')
d2=pd.read_stata('oregonhie_ed_vars.dta')
d3=pd.read_stata('oregonhie_stateprograms_vars.dta')
d4=pd.merge(d1,d2,how='left',on=['person_id'])

In [3]:
data1=pd.merge(d3,d4,how='left',on=['person_id'])

In [4]:
data2=pd.get_dummies(data1,columns=['numhh_list'])

In [5]:
data3=data2.copy()

## (a)

**Balance check inPortland**

In [6]:
def t(x):
    if x == 'Selected':
        x = 1
    else:
        x = 0
    return x

In [7]:
#transfer treatment into 0-1
data3['treatment']=data2['treatment'].apply(lambda x: t(x))

In [8]:
#inPorland
inPortland=[]
sign = np.array(data3['any_loun_ed'])
for i in range(len(sign)):
    if sign[i]=='Yes':
        inPortland.append(1)
    elif sign[i] == 'No':
        inPortland.append(1)
    else: inPortland.append(0)
data3['inPortland'] = inPortland

In [9]:
LinearModel_new_spending = sm.OLS(endog = data3['inPortland'], exog = sm.add_constant(data3[['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
          'numhh_list_signed self up + 2 additional people']]))
result_new_spending = LinearModel_new_spending.fit()
print(result_new_spending.summary())

                            OLS Regression Results                            
Dep. Variable:             inPortland   R-squared:                       0.002
Model:                            OLS   Adj. R-squared:                  0.002
Method:                 Least Squares   F-statistic:                     59.02
Date:                Mon, 07 Mar 2022   Prob (F-statistic):           4.25e-38
Time:                        19:10:57   Log-Likelihood:                -49627.
No. Observations:               74922   AIC:                         9.926e+04
Df Residuals:                   74918   BIC:                         9.930e+04
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                                                      coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------

**The P>|t| of treatment is 0.769 larger than 0.05, so the data is balanced.**

**Balance check birthyear_list**

In [10]:
data4=data3.copy()

In [11]:
data4[data4['inPortland']==1]['birthyear_list']

4        1969
7        1968
8        1977
15       1971
17       1957
         ... 
74906    1953
74910    1963
74914    1951
74917    1955
74919    1965
Name: birthyear_list, Length: 24646, dtype: int16

In [12]:
features=['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
          'numhh_list_signed self up + 2 additional people']
LinearModel_new_spending = sm.OLS(endog = data4[data4['inPortland']==1]['birthyear_list'], exog = sm.add_constant(data4[data4['inPortland']==1][['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
      'numhh_list_signed self up + 2 additional people']]))
result_new_spending = LinearModel_new_spending.fit()
print(result_new_spending.summary())

                            OLS Regression Results                            
Dep. Variable:         birthyear_list   R-squared:                      -0.001
Model:                            OLS   Adj. R-squared:                 -0.001
Method:                 Least Squares   F-statistic:                    -4.992
Date:                Mon, 07 Mar 2022   Prob (F-statistic):               1.00
Time:                        19:10:58   Log-Likelihood:                -96317.
No. Observations:               24646   AIC:                         1.926e+05
Df Residuals:                   24641   BIC:                         1.927e+05
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                                                      coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------

**The P>|t| of treatment is 0.530, which is larger than 0.05, implying that the data is balanced.**

In [13]:
data5=data4.copy()

In [14]:
def gender(x):
    if x == '1: Female':
        x = 1
    else:
        x = 0
    return x

In [15]:
data5['female_list']=data4['female_list'].apply(lambda x: gender(x))

In [16]:
LinearModel_new_spending = sm.OLS(endog = data5[data5['inPortland']==1]['female_list'], exog = sm.add_constant(data5[data5['inPortland']==1][['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
      'numhh_list_signed self up + 2 additional people']]))
result_new_spending = LinearModel_new_spending.fit()
print(result_new_spending.summary())

                            OLS Regression Results                            
Dep. Variable:            female_list   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.000
Method:                 Least Squares   F-statistic:                     3.988
Date:                Mon, 07 Mar 2022   Prob (F-statistic):            0.00309
Time:                        19:10:58   Log-Likelihood:                -17772.
No. Observations:               24646   AIC:                         3.555e+04
Df Residuals:                   24641   BIC:                         3.559e+04
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                                                      coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------

**The P>|t| of treatment is 0.144, which is larger than 0.05, implying that the data is balanced.**

In [17]:
data6=data5.copy()

In [18]:
def signself(x):
    if x == 'Signed self up':
        x = 1
    else:
        x=0
    return x

In [19]:
data6['self_list']=data5['self_list'].apply(lambda x: signself(x))

In [20]:
LinearModel_new_spending = sm.OLS(endog = data6[data6['inPortland']==1]['self_list'],
                                  exog = sm.add_constant(data6[data6['inPortland']==1][['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
    'numhh_list_signed self up + 2 additional people']]))
result_new_spending = LinearModel_new_spending.fit()
print(result_new_spending.summary())

                            OLS Regression Results                            
Dep. Variable:              self_list   R-squared:                       0.428
Model:                            OLS   Adj. R-squared:                  0.428
Method:                 Least Squares   F-statistic:                     4603.
Date:                Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                        19:10:58   Log-Likelihood:                 1374.9
No. Observations:               24646   AIC:                            -2740.
Df Residuals:                   24641   BIC:                            -2699.
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                                                      coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------

**The P>|t| of treatment is 0.831, which is larger than 0.05, implying that the data is balanced.**

Balance check for 'any_visit_pre_ed'

In [21]:
data7=data6.copy()

In [22]:
def yn(x):
    if x=='Yes':
        x=1
    else:
        x=0
    return x

In [23]:
data7['any_visit_pre_ed']=data6['any_visit_pre_ed'].apply(lambda x: yn(x))

In [24]:
data7[data7['inPortland']==1]['any_visit_pre_ed'].value_counts()

0    16930
1     7716
Name: any_visit_pre_ed, dtype: int64

In [25]:
LinearModel_new_spending = sm.OLS(endog = data7[data7['inPortland']==1]['any_visit_pre_ed'], 
                                  exog = sm.add_constant(data7[data7['inPortland']==1][['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
                                  'numhh_list_signed self up + 2 additional people']]))
result_new_spending = LinearModel_new_spending.fit()
print(result_new_spending.summary())

                            OLS Regression Results                            
Dep. Variable:       any_visit_pre_ed   R-squared:                       0.015
Model:                            OLS   Adj. R-squared:                  0.015
Method:                 Least Squares   F-statistic:                     92.21
Date:                Mon, 07 Mar 2022   Prob (F-statistic):           5.82e-78
Time:                        19:10:58   Log-Likelihood:                -15850.
No. Observations:               24646   AIC:                         3.171e+04
Df Residuals:                   24641   BIC:                         3.175e+04
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
                                                      coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------------------

**The P>|t| of treatment is 0.461, which is larger than 0.05, implying that the data is balanced.**

In [26]:
data8=data7[data7['inPortland']==1][['num_visit_pre_cens_ed',
                                     'treatment',
                                     'numhh_list_signed self up',
                                     'numhh_list_signed self up + 1 additional person',
                                     'numhh_list_signed self up + 2 additional people']].dropna()

In [27]:
LinearModel_new_spending = sm.OLS(endog = data8['num_visit_pre_cens_ed'], 
                                  exog = sm.add_constant(data8[['treatment','numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
                                 'numhh_list_signed self up + 2 additional people']]))
result_new_spending = LinearModel_new_spending.fit()
print(result_new_spending.summary())

                              OLS Regression Results                             
Dep. Variable:     num_visit_pre_cens_ed   R-squared:                       0.010
Model:                               OLS   Adj. R-squared:                  0.009
Method:                    Least Squares   F-statistic:                     59.14
Date:                   Mon, 07 Mar 2022   Prob (F-statistic):           8.92e-50
Time:                           19:10:58   Log-Likelihood:                -50165.
No. Observations:                  24634   AIC:                         1.003e+05
Df Residuals:                      24629   BIC:                         1.004e+05
Df Model:                              4                                         
Covariance Type:               nonrobust                                         
                                                      coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------

**The P>|t| of treatment is 0.928, which is larger than 0.05, implying that the data is balanced.**

## (b)

In [28]:
data9=data7[data7['inPortland']==1]

Y='ohp_all_ever_firstn_30sep2009'\
W='treatment'\
X=df_feature

**variable selection**

Here are the first edition of feature candidates, not the final version.\
1.numhh_list (To be confirmed.)\
2.birthyear_list (This variable is correlated to age. To be confirmed.)\
3.female_list (Gender should have correlation to the selection. To be confirmed.)\
4.any_visit_ed (If one has been to ED, he/she would be more careful about their own health. To be confirmed.)\
5.num_visit_cens_ed (If one has been to ED more than others, he/she would be more likely to be selected. To be confirmed.)\
6.any_hosp_ed (If one has been to ED, and be hospitalized he/she would be more careful about their own health. To be confirmed.)\
7.any_out_ed(Been to ED but outpatient. To be confirmed.)\

In [29]:
df_feature = data9[['numhh_list_signed self up','numhh_list_signed self up + 1 additional person','numhh_list_signed self up + 2 additional people','birthyear_list','female_list','any_visit_ed',\
'num_visit_cens_ed','any_hosp_ed','any_out_ed','ohp_all_ever_firstn_30sep2009','treatment']]
df_dummy = pd.get_dummies(df_feature,columns=['any_visit_ed','any_hosp_ed','any_out_ed','ohp_all_ever_firstn_30sep2009'],drop_first=True).dropna()
df_dummy

numhh_list_signed self up  \
4                              1   
7                              0   
8                              1   
15                             0   
17                             1   
...                          ...   
74906                          1   
74910                          0   
74914                          1   
74917                          1   
74919                          1   

       numhh_list_signed self up + 1 additional person  \
4                                                    0   
7                                                    1   
8                                                    0   
15                                                   1   
17                                                   0   
...                                                ...   
74906                                                0   
74910                                                1   
74914                                                0   
74917                                                0   
74919                                                0   

       numhh_list_signed self up + 2 additional people  birthyear_list  \
4                                                    0            1969   
7                                                    0            1968   
8                                                    0            1977   
15                                                   0            1971   
17                                                   0            1957   
...                                                ...             ...   
74906                                                0            1953   
74910                                                0            1963   
74914                                                0            1951   
74917                                                0            1955   
74919                                                0            1965   

      female_list  num_visit_cens_ed treatment  any_visit_ed_Yes  \
4               1                0.0         1                 0   
7               0                2.0         0                 1   
8               1                0.0         0                 0   
15              1                5.0         0                 1   
17              0                0.0         0                 0   
...           ...                ...       ...               ...   
74906           0                0.0         1                 0   
74910           0                0.0         1                 0   
74914           1                0.0         1                 0   
74917           0                5.0         0                 1   
74919           1                0.0         0                 0   

       any_hosp_ed_Yes  any_out_ed_Yes  ohp_all_ever_firstn_30sep2009_Enrolled  
4                    0               0                                       0  
7                    0               1                                       0  
8                    0               0                                       0  
15                   1               1                                       1  
17                   0               0                                       0  
...                ...             ...                                     ...  
74906                0               0                                       0  
74910                0               0                                       0  
74914                0               0                                       1  
74917                0               1                                       0  
74919                0               0                                       1  

[24622 rows x 11 columns]

In [30]:
feature = ['numhh_list_signed self up','numhh_list_signed self up + 1 additional person',
           'numhh_list_signed self up + 2 additional people','birthyear_list','female_list',
           'num_visit_cens_ed',\
            'any_visit_ed_Yes','any_hosp_ed_Yes','any_out_ed_Yes','treatment']
for i in range(len(feature)):
    BalanceCheck = sm.OLS(endog = df_dummy[feature[i]].astype(float), exog = sm.add_constant(df_dummy['treatment']))
    result_BC = BalanceCheck.fit()
    print(result_BC.summary())

                                OLS Regression Results                               
Dep. Variable:     numhh_list_signed self up   R-squared:                       0.036
Model:                                   OLS   Adj. R-squared:                  0.036
Method:                        Least Squares   F-statistic:                     920.0
Date:                       Mon, 07 Mar 2022   Prob (F-statistic):          1.98e-198
Time:                               19:10:58   Log-Likelihood:                -12065.
No. Observations:                      24622   AIC:                         2.413e+04
Df Residuals:                          24620   BIC:                         2.415e+04
Df Model:                                  1                                         
Covariance Type:                   nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------

According the balance check, ['birthyear_list], ['num_visit_cens_ed'], ['any_visit_ed_Yes'],['any_out_ed_Yes] are balanced. 

**Proof of  number of household:**

In [31]:
#When we do not delete ['numhh_list_signed self up + 2 additional people']:
BalanceCheck = sm.OLS(endog = df_dummy['ohp_all_ever_firstn_30sep2009_Enrolled'].astype(float), exog = sm.add_constant(df_dummy[['numhh_list_signed self up','numhh_list_signed self up + 1 additional person','numhh_list_signed self up + 2 additional people','birthyear_list','female_list','num_visit_cens_ed',\
                                                                                                                                 'any_visit_ed_Yes','any_hosp_ed_Yes','any_out_ed_Yes','treatment']]))
result_BC = BalanceCheck.fit()
print(result_BC.summary())

                                      OLS Regression Results                                      
Dep. Variable:     ohp_all_ever_firstn_30sep2009_Enrolled   R-squared:                       0.112
Model:                                                OLS   Adj. R-squared:                  0.112
Method:                                     Least Squares   F-statistic:                     344.6
Date:                                    Mon, 07 Mar 2022   Prob (F-statistic):               0.00
Time:                                            19:10:58   Log-Likelihood:                -12696.
No. Observations:                                   24622   AIC:                         2.541e+04
Df Residuals:                                       24612   BIC:                         2.549e+04
Df Model:                                               9                                         
Covariance Type:                                nonrobust                                         
          

There are two reasons why the number of household **should not be deleted:** \
Reason 1.From the rules we know that: **Those in a larger household are more likely to be selected into the treatment condition. If we delete the number of household, the causal effect of other variables would be overestimated.**\
Reason 2.The **coefficients** of **'numhh_list_signed self up '**, **'numhh_list_signed self up + 1 addition person'** and **'numhh_list_signed self up + 2 additional people'** are respectively **-0.3264, -0.3357, -0.5189.** The reasonable interpretation of this is: The more households a person sign up, the more possible that s/he did not enroll in any medical program in the period of March 10th, 2008 to Sept 30th, 2009.

**By controlling the features, our regression results imply that:**

**being selected by lottery increases the probability that an individual was enrolled in any Medicaid Program during the study period by 24.45% on average**

**We have 95% confidence that being selected by lottery increases the probability an individual was enrolled in any any Medicaid Program during the study period by ( 23.4% , 25.5% )**

**Final edition of features:**\
1.numhh_list(confirmed)\
2.birthyear_list (Confirmed.)\
3.female_list (Confirmed.)\
4.num_visit_cens_ed (Confirmed.)\
5.any_visit_ed_Yes (Confirmed)\
6.any_hosp_ed (Confirmed.)\
7.any_out_ed(Confirmed.)

## (C)

W='ohp_all_ever_firstn_30sep2009'(Ever enrolled in Medicaid from 1st not. date (10mar2008) to 30sep2009)\
Y1='any_visit_ed'(Any ED visit in the study period)\
Y2='num_visit_cens_ed'(Number of ED visits in the study period (Censored))

## 1. Check the causal effect of Enrolling into a Medicaid Program on ANY ED VISITS using IV2SLS

In [32]:
data10=data1.copy()
data10['inPortland'] = inPortland

As for the features X, I choose those with binary variables related to the ED utilization\
1:'postn_death'(Death post notification date (2008 and 2009): Oregon Vital 74922 Statistics data)\
2:'birthyear_list'(Birth year: lottery list data)\
3:'have_phone_list'(Gave a phone number on lottery sign up)\
4:'english_list'(Individual requested english-language materials)\
5:'female_list'(Female)\
6:'any_visit_pre_ed'(Any ED visit, pre-randomization)\
7:'num_visit_pre_cens_ed'(Number of of ED visits, pre-randomization (Censored))\

In [33]:
data11=data10[data10['inPortland']==1][['postn_death','birthyear_list','have_phone_list','english_list',
                       'female_list','any_visit_ed','any_visit_pre_ed','num_visit_pre_cens_ed',
                       'ohp_all_ever_firstn_30sep2009','treatment','numhh_list']]

In [34]:
data11

postn_death  birthyear_list    have_phone_list  \
4           Alive            1969  Gave Phone Number   
7           Alive            1968  Gave Phone Number   
8           Alive            1977  Gave Phone Number   
15          Alive            1971  Gave Phone Number   
17          Alive            1957  Gave Phone Number   
...           ...             ...                ...   
74906       Alive            1953  Gave Phone Number   
74910       Alive            1963  Gave Phone Number   
74914       Alive            1951  Gave Phone Number   
74917       Alive            1955  Gave Phone Number   
74919       Alive            1965  Gave Phone Number   

                                            english_list female_list  \
4      Requested materials in language other than eng...   1: Female   
7                            Requested English materials     0: Male   
8                            Requested English materials   1: Female   
15                           Requested English materials   1: Female   
17                           Requested English materials     0: Male   
...                                                  ...         ...   
74906                        Requested English materials     0: Male   
74910                        Requested English materials     0: Male   
74914  Requested materials in language other than eng...   1: Female   
74917                        Requested English materials     0: Male   
74919                        Requested English materials   1: Female   

      any_visit_ed any_visit_pre_ed  num_visit_pre_cens_ed  \
4               No               No                    0.0   
7              Yes               No                    0.0   
8               No              Yes                    1.0   
15             Yes              Yes                    1.0   
17              No              Yes                    2.0   
...            ...              ...                    ...   
74906           No               No                    0.0   
74910           No               No                    0.0   
74914           No               No                    0.0   
74917          Yes               No                    0.0   
74919           No              Yes                    1.0   

      ohp_all_ever_firstn_30sep2009     treatment  \
4                      NOT enrolled      Selected   
7                      NOT enrolled  Not selected   
8                      NOT enrolled  Not selected   
15                         Enrolled  Not selected   
17                     NOT enrolled  Not selected   
...                             ...           ...   
74906                  NOT enrolled      Selected   
74910                  NOT enrolled      Selected   
74914                      Enrolled      Selected   
74917                  NOT enrolled  Not selected   
74919                      Enrolled  Not selected   

                                 numhh_list  
4                            signed self up  
7      signed self up + 1 additional person  
8                            signed self up  
15     signed self up + 1 additional person  
17                           signed self up  
...                                     ...  
74906                        signed self up  
74910  signed self up + 1 additional person  
74914                        signed self up  
74917                        signed self up  
74919                        signed self up  

[24646 rows x 11 columns]

In [35]:
data12=pd.get_dummies(data11,columns=['postn_death','have_phone_list','english_list',
                       'female_list','any_visit_ed','any_visit_pre_ed',
                       'ohp_all_ever_firstn_30sep2009','treatment','numhh_list'],drop_first=True).dropna()

In [36]:
data12

birthyear_list  num_visit_pre_cens_ed  postn_death_Dead  \
4                1969                    0.0                 0   
7                1968                    0.0                 0   
8                1977                    1.0                 0   
15               1971                    1.0                 0   
17               1957                    2.0                 0   
...               ...                    ...               ...   
74906            1953                    0.0                 0   
74910            1963                    0.0                 0   
74914            1951                    0.0                 0   
74917            1955                    0.0                 0   
74919            1965                    1.0                 0   

       have_phone_list_Gave Phone Number  \
4                                      1   
7                                      1   
8                                      1   
15                                     1   
17                                     1   
...                                  ...   
74906                                  1   
74910                                  1   
74914                                  1   
74917                                  1   
74919                                  1   

       english_list_Requested English materials  female_list_1: Female  \
4                                             0                      1   
7                                             1                      0   
8                                             1                      1   
15                                            1                      1   
17                                            1                      0   
...                                         ...                    ...   
74906                                         1                      0   
74910                                         1                      0   
74914                                         0                      1   
74917                                         1                      0   
74919                                         1                      1   

       any_visit_ed_Yes  any_visit_pre_ed_Yes  \
4                     0                     0   
7                     1                     0   
8                     0                     1   
15                    1                     1   
17                    0                     1   
...                 ...                   ...   
74906                 0                     0   
74910                 0                     0   
74914                 0                     0   
74917                 1                     0   
74919                 0                     1   

       ohp_all_ever_firstn_30sep2009_Enrolled  treatment_Selected  \
4                                           0                   1   
7                                           0                   0   
8                                           0                   0   
15                                          1                   0   
17                                          0                   0   
...                                       ...                 ...   
74906                                       0                   1   
74910                                       0                   1   
74914                                       1                   1   
74917                                       0                   0   
74919                                       1                   0   

       numhh_list_signed self up + 1 additional person  \
4                                                    0   
7                                                    1   
8                                                    0   
15                                                   1   
17                                                   0   
...                           

Balance check

In [37]:
feature1 = ['birthyear_list','female_list_1: Female','postn_death_Dead','have_phone_list_Gave Phone Number',
                                     'english_list_Requested English materials',
                                     'numhh_list_signed self up + 1 additional person','numhh_list_signed self up + 2 additional people']
for i in range(len(feature1)):
    BalanceCheck = sm.OLS(endog = data12[feature1[i]].astype(float), exog = sm.add_constant(data12['ohp_all_ever_firstn_30sep2009_Enrolled']))
    result_BC = BalanceCheck.fit()
    print(result_BC.summary())

                            OLS Regression Results                            
Dep. Variable:         birthyear_list   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     13.56
Date:                Mon, 07 Mar 2022   Prob (F-statistic):           0.000232
Time:                        19:10:58   Log-Likelihood:                -96255.
No. Observations:               24634   AIC:                         1.925e+05
Df Residuals:                   24632   BIC:                         1.925e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                                             coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------

According to the balance check, the estimation is **biased**. Directly using OLS may not be appropriate.

**We use IV2Sl to acquire an unbiased estimation.**

In [38]:
# Fit a 2sls model.
import linearmodels
from linearmodels.iv import IV2SLS
data12['const']=1
IV_model = IV2SLS(dependent = data12['any_visit_ed_Yes'] ,endog = data12['ohp_all_ever_firstn_30sep2009_Enrolled'],\
                  exog = data12[['birthyear_list','female_list_1: Female','postn_death_Dead','have_phone_list_Gave Phone Number',
                                     'english_list_Requested English materials',
                                     'numhh_list_signed self up + 1 additional person','numhh_list_signed self up + 2 additional people','const']],
                  instruments=data12['treatment_Selected'])
res_2sls = IV_model.fit()
print(res_2sls)

                          IV-2SLS Estimation Summary                          
Dep. Variable:       any_visit_ed_Yes   R-squared:                      0.0512
Estimator:                    IV-2SLS   Adj. R-squared:                 0.0509
No. Observations:               24634   F-statistic:                    1359.3
Date:                Mon, Mar 07 2022   P-value (F-stat)                0.0000
Time:                        19:10:59   Distribution:                  chi2(8)
Cov. Estimator:                robust                                         
                                                                              
                                                Parameter Estimates                                                
                                                 Parameter  Std. Err.     T-stat    P-value    Lower CI    Upper CI
-------------------------------------------------------------------------------------------------------------------
birthyear_list      

**By controlling the features, our regression results imply that:**

**being enrolled into a Medicaid Program increases the probability that an individual visit Emergency Department at least once. during the study period by 7.01% on average**.

**We have 95% confidence that being enrolled into a Medicaid Program will increase the probability that an individual visit Emergency Department at least once during the study period by ( 2.15% , 11.87% )**

## Now Test Y2='num_visit_cens_ed', W=ohp_all_ever_firstn_30sep2009

As for the features X, I choose those with binary variables related to the ED utilization\
1:'postn_death'(Death post notification date (2008 and 2009): Oregon Vital 74922 Statistics data)\
2:'birthyear_list'(Birth year: lottery list data)\
3:'have_phone_list'(Gave a phone number on lottery sign up)\
4:'english_list'(Individual requested english-language materials)\
5:'female_list'(Female)\
6:'any_visit_pre_ed'(Any ED visit, pre-randomization)\
7:'num_visit_pre_cens_ed'(Number of of ED visits, pre-randomization (Censored))\

get dummy

In [39]:
data13 = data1.copy()
data13['inPortland']=inPortland

In [40]:
data14=data13[data13['inPortland']==1][['postn_death','birthyear_list','have_phone_list','english_list','female_list','any_visit_pre_ed','num_visit_pre_cens_ed','numhh_list','num_visit_cens_ed','treatment','ohp_all_ever_firstn_30sep2009']]

In [41]:
data15=pd.get_dummies(data14,columns=['postn_death','have_phone_list','english_list','female_list','any_visit_pre_ed','numhh_list','treatment','ohp_all_ever_firstn_30sep2009'],drop_first=True)

In [42]:
data15

birthyear_list  num_visit_pre_cens_ed  num_visit_cens_ed  \
4                1969                    0.0                0.0   
7                1968                    0.0                2.0   
8                1977                    1.0                0.0   
15               1971                    1.0                5.0   
17               1957                    2.0                0.0   
...               ...                    ...                ...   
74906            1953                    0.0                0.0   
74910            1963                    0.0                0.0   
74914            1951                    0.0                0.0   
74917            1955                    0.0                5.0   
74919            1965                    1.0                0.0   

       postn_death_Dead  have_phone_list_Gave Phone Number  \
4                     0                                  1   
7                     0                                  1   
8                     0                                  1   
15                    0                                  1   
17                    0                                  1   
...                 ...                                ...   
74906                 0                                  1   
74910                 0                                  1   
74914                 0                                  1   
74917                 0                                  1   
74919                 0                                  1   

       english_list_Requested English materials  female_list_1: Female  \
4                                             0                      1   
7                                             1                      0   
8                                             1                      1   
15                                            1                      1   
17                                            1                      0   
...                                         ...                    ...   
74906                                         1                      0   
74910                                         1                      0   
74914                                         0                      1   
74917                                         1                      0   
74919                                         1                      1   

       any_visit_pre_ed_Yes  numhh_list_signed self up + 1 additional person  \
4                         0                                                0   
7                         0                                                1   
8                         1                                                0   
15                        1                                                1   
17                        1                                                0   
...                     ...                                              ...   
74906                     0                                                0   
74910                     0                                                1   
74914                     0                                                0   
74917                     0                                                0   
74919                     1                                                0   

       numhh_list_signed self up + 2 additional people  treatment_Selected  \
4                                                    0                   1   
7                                                    0                   0   
8                                                    0                   0   
15                                                   0                   0   
17                                                   0                   0   
...                                                ...                 ...   
74906                                                0       

Balance Check

In [43]:
feature2 = ['birthyear_list','postn_death_Dead', 'have_phone_list_Gave Phone Number',
       'english_list_Requested English materials', 'female_list_1: Female',
       'numhh_list_signed self up + 1 additional person',
       'numhh_list_signed self up + 2 additional people',]
for i in range(len(feature2)):
    BalanceCheck = sm.OLS(endog = data15[feature2[i]].astype(float), exog = sm.add_constant(data15['ohp_all_ever_firstn_30sep2009_Enrolled']))
    result_BC = BalanceCheck.fit()
    print(result_BC.summary())

                            OLS Regression Results                            
Dep. Variable:         birthyear_list   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     13.63
Date:                Mon, 07 Mar 2022   Prob (F-statistic):           0.000223
Time:                        19:10:59   Log-Likelihood:                -96300.
No. Observations:               24646   AIC:                         1.926e+05
Df Residuals:                   24644   BIC:                         1.926e+05
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                                             coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------------------------

According to the balance check, the estimation is **biased** estimation.

Therefore, conduct an IV2SLS analysis to acquire an unbias estimation.

In [44]:
# Fit a 2sls model.
import linearmodels
from linearmodels.iv import IV2SLS
data15['const']=1
IV_model = IV2SLS(dependent = data15['num_visit_cens_ed'] ,endog = data15['ohp_all_ever_firstn_30sep2009_Enrolled'],\
                  exog = data15[['birthyear_list', 'num_visit_pre_cens_ed',
       'postn_death_Dead', 'have_phone_list_Gave Phone Number',
       'english_list_Requested English materials', 'female_list_1: Female',
       'any_visit_pre_ed_Yes',
       'numhh_list_signed self up + 1 additional person',
       'numhh_list_signed self up + 2 additional people','const']],
                  instruments=data15['treatment_Selected'])
res_2sls = IV_model.fit()
print(res_2sls)

                          IV-2SLS Estimation Summary                          
Dep. Variable:      num_visit_cens_ed   R-squared:                      0.3422
Estimator:                    IV-2SLS   Adj. R-squared:                 0.3419
No. Observations:               24615   F-statistic:                    3006.8
Date:                Mon, Mar 07 2022   P-value (F-stat)                0.0000
Time:                        19:10:59   Distribution:                 chi2(10)
Cov. Estimator:                robust                                         
                                                                              
                                                Parameter Estimates                                                
                                                 Parameter  Std. Err.     T-stat    P-value    Lower CI    Upper CI
-------------------------------------------------------------------------------------------------------------------
birthyear_list      

/Users/mac/opt/anaconda3/lib/python3.8/site-packages/linearmodels/shared/exceptions.py:37: MissingValueWarning: 
Inputs contain missing values. Dropping rows with missing observations.
  warnings.warn(missing_value_warning_msg, MissingValueWarning)


**By controlling the features, our regression results imply that:**

**Being enrolled into a Medicaid Program during study period increases the times that an individual visit Emergency Department during the study period by 0.3788 times on average**.

**We have 95% confidence that being enrolled into a Medicaid Program will increase the times that an individual visit Emergency Department during the study period by (0.1717 , 0.5859) times.**